extensions/amp-story/1.0/page-advancement.js

```diff
@@ -796,6 +821,11 @@ export class TimeBasedAdvancement extends AdvancementConfig {
     }
   }
 
+  /** @override */
+  getType() {
+    return 'TimeBasedAdvancement';
```

Comment:
These `getType` could work much better as enums, and you also save bundle-size because the actual values would be numbers and not long strings.